src/client/compound/compound-bridge-data.ts

```diff
@@ -0,0 +1,150 @@
+  }
+
+  async getExpectedOutput(
+    inputAssetA: AztecAsset,
+    inputAssetB: AztecAsset,
+    outputAssetA: AztecAsset,
+    outputAssetB: AztecAsset,
+    auxData: bigint,
+    inputValue: bigint,
+  ): Promise<bigint[]> {
+    if (auxData === 0n) {
+      // Minting
+      const cToken = ICERC20__factory.connect(outputAssetA.erc20Address, this.ethersProvider);
+      const exchangeRateStored = await cToken.exchangeRateStored();
+      return [BigNumber.from(inputValue).mul(this.expScale).div(exchangeRateStored).toBigInt()];
+    } else if (auxData === 1n) {
+      // Redeeming
+      const cToken = ICERC20__factory.connect(inputAssetA.erc20Address, this.ethersProvider);
+      const exchangeRateStored = await cToken.exchangeRateStored();
+      return [BigNumber.from(inputValue).mul(exchangeRateStored).div(this.expScale).toBigInt()];
+    } else {
+      throw "Invalid auxData";
+    }
+  }
+
+  async getExpectedYield(
+    inputAssetA: AztecAsset,
+    inputAssetB: AztecAsset,
+    outputAssetA: AztecAsset,
+    outputAssetB: AztecAsset,
+    auxData: bigint,
+    precision: bigint,
+  ): Promise<number[]> {
+    // Not taking into account how the deposited funds will change the yield
+    if (auxData === 0n) {
+      // Minting
+      // The approximate number of blocks per year that is assumed by the interest rate model
+      const blocksPerYear = 2102400;
+      const supplyRatePerBlock = await ICERC20__factory.connect(
+        outputAssetA.erc20Address,
+        this.ethersProvider,
+      ).supplyRatePerBlock();
+      return [supplyRatePerBlock.mul(blocksPerYear).toNumber() / 10 ** 16];
+    } else if (auxData === 1n) {
+      // Redeeming
+      return [0];
+    } else {
+      throw "Invalid auxData";
+    }
+  }
+
+  async getMarketSize(
+    inputAssetA: AztecAsset,
+    inputAssetB: AztecAsset,
+    outputAssetA: AztecAsset,
+    outputAssetB: AztecAsset,
+    auxData: bigint,
+  ): Promise<AssetValue[]> {
+    let cTokenAddress;
+    let underlyingAsset;
+    if (auxData === 0n) {
+      // Minting
+      cTokenAddress = outputAssetA.erc20Address;
+      underlyingAsset = inputAssetA;
+    } else if (auxData === 1n) {
+      // Redeeming
+      cTokenAddress = inputAssetA.erc20Address;
+      underlyingAsset = outputAssetA;
+    } else {
+      throw "Invalid auxData";
+    }
+
+    let marketSize;
+    if (underlyingAsset.assetType === AztecAssetType.ETH) {
+      marketSize = await this.ethersProvider.getBalance(cTokenAddress);
+    } else {
+      marketSize = await IERC20__factory.connect(underlyingAsset.erc20Address, this.ethersProvider).balanceOf(
+        cTokenAddress,
+      );
+    }
+    return [
+      {
+        assetId: underlyingAsset.id,
+        amount: marketSize.toBigInt(),
+      },
+    ];
+  }
+
+  private async getAllMarkets(): Promise<EthAddress[]> {
+    if (!this.allMarkets) {
+      const allMarketsString = await IComptroller__factory.connect(
+        "0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B",
```

Comment:
Would set the address as a constant higher up, easier to spot it and not be confused what it is etc.